x-pack/plugins/maps/public/classes/styles/vector/components/style_prop_editor.tsx

```diff
@@ -14,12 +13,31 @@ import {
   EuiFieldText,
   EuiToolTip,
 } from '@elastic/eui';
-import { STYLE_TYPE } from '../../../../../common/constants';
 import { i18n } from '@kbn/i18n';
+import { getVectorStyleLabel, getDisabledByMessage } from './get_vector_style_label';
+import { STYLE_TYPE, VECTOR_STYLES } from '../../../../../common/constants';
+import { FieldMetaOptions } from '../../../../../common/descriptor_types';
+import { IStyleProperty } from '../properties/style_property';
+import { StyleField } from '../style_fields_helper';
+
+export interface Props<StaticOptionsType, DynamicOptionsType> {
```

Comment:
nit: do we need the `Type`-suffix? The compiler will enforce that it's.a type and not an instance.

e.g.:

```Props<StaticOptions, DynamicOptions>```

Reply:
I added the `Type` suffix to make the names more explicit on what they are. Once you start adding angle brackets everywhere the names blur together so I figured its more straightforward that they are types and not something other thing. I can remove if needed.